enterprise/course_catalog_api.py

```diff
@@ -158,20 +160,23 @@ def get_common_course_modes(self, course_run_ids):
 
         return available_course_modes
 
-    def _load_data(self, resource, **kwargs):
+    def _load_data(self, resource, default=DEFAULT_VALUE_SAFEGUARD, **kwargs):
         """
         Load data from API client.
 
         Arguments:
-            resource(str): type of resource to load
+            resource(string): type of resource to load
+            default(any): value to return if API query returned empty result. Sensible values: [], {}, None etc.
 
         Returns:
             dict: Deserialized response from Course Catalog API
         """
-        return get_edx_api_data(
+        default_val = default if default != self.DEFAULT_VALUE_SAFEGUARD else {}
```

Comment:
Why have `DEFAULT_VALUE_SAFEGUARD` at all if we will return `{}` when it is detected?